### PR TITLE
PADV-1480 feat: enroll student modal

### PR DESCRIPTION
**Pull request by AuraAlba**
### Ticket
https://agile-jira.pearson.com/browse/PADV-1480

### Description
Add enroll student modal

### Changes made
Add enroll student modal
Add api to get messages and enroll student
Add styles for modal
Add test

### Screenshots
Screenshot from 2024-08-09 22-55-06

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1990/classes -> class detail

### CODE REVIEW

File: src/features/Classes/data/api.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { getConfig } from '@edx/frontend-platform';
+
+function handleEnrollments(data, courseId) {
+  const INSTRUCTOR_API_URL = `${getConfig().LMS_BASE_URL}/courses/course_id/instructor/api`;
+  const courseIdSearchPattern = /course_id/;
+
+  return getAuthenticatedHttpClient().post(
+    `${INSTRUCTOR_API_URL.replace(courseIdSearchPattern, courseId)}/students_update_enrollment`,
+    data,
+  );
```
Comment (01001110J) [inline src/features/Classes/data/api.js:11]: What about this:

```suggestion
  const ENROLLMENT_API_URL = `${getConfig().LMS_BASE_URL}/courses/${courseId}/instructor/api/students_update_enrollment`;

  return getAuthenticatedHttpClient().post(
     ENROLLMENT_API_URL,
    data,
  );
```

File: src/features/Classes/EnrollStudent/index.jsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                  name="studentEmail"
+                  required
+                  autoResize
+                />
+              </FormGroup>
+              <div className="d-flex justify-content-end">
+                <ModalCloseButton className="btntpz btn-text btn-tertiary mr-2">
+                  Cancel
+                </ModalCloseButton>
+                <Button type="submit">Send invite</Button>
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:142]: I suggest to leave this button disabled if the field is empty or its value is minor to idk 10?

File: src/features/Classes/EnrollStudent/index.jsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          {!isLoading && (
+            <Form onSubmit={handleEnrollStudent}>
+              <FormGroup controlId="studentInfo">
+                <Form.Control
+                  as="textarea"
+                  placeholder="Enter email of the student to enroll"
+                  floatingLabel="Email"
+                  className="my-4 mr-0 student-email"
+                  name="studentEmail"
+                  required
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:134]: I could avoid this restriction just with a single `.` I suggest to verify if the email is an valid email before make the request

image

File: src/features/Classes/EnrollStudent/index.jsx (hunk 2 above)
Comment (AuraAlba) [reply src/features/Classes/EnrollStudent/index.jsx:142]: I leave the button disabled if the field have empty characters, but this api received also usernames so I can't limit the length

File: src/features/Classes/EnrollStudent/index.jsx (hunk 3 above)
Comment (AuraAlba) [reply src/features/Classes/EnrollStudent/index.jsx:134]: As I said in the previous comment this api accepts also usernames, also in frontend we can't cover all the cases to say if is an email valid that the reason backend is in charge of the validation.

**Review by 01001110J — APPROVED** [review]

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**